core/keyserver/validate.go

```diff
@@ -100,6 +103,31 @@ func validateListEntryHistoryRequest(in *pb.ListEntryHistoryRequest, currentRevi
 	return nil
 }
 
+// validateListUserRevisionsRequest checks the bounds on start and end revisions and returns an appropriate number of
+// revisions to return for this request.
+func validateListUserRevisionsRequest(in *pb.ListUserRevisionsRequest, offset, newestRevision int64) (int64, error) {
+	if in.StartRevision < 0 || in.StartRevision > newestRevision {
+		return 0, ErrInvalidStart
+	}
+	if in.EndRevision < in.StartRevision || in.EndRevision > newestRevision {
+		return 0, ErrInvalidEnd
+	}
+
+	revisions := int64(in.PageSize)
+	switch {
+	case in.PageSize < 0:
+		return 0, fmt.Errorf("Invalid page size")
+	case in.PageSize == 0:
+		revisions = int64(defaultPageSize)
+	case in.PageSize > maxPageSize:
+		revisions = int64(maxPageSize)
+	}
+	if in.StartRevision+offset+revisions > in.EndRevision {
+		revisions = in.EndRevision - (in.StartRevision + offset) + 1
```

Comment:
it seems like you mostly make use of in.StartRevision + offset rather than either of them alone. Should we just store them in a "pageStart" variable to reduce the number of terms?

Reply:
Done.